### ISSUE TITLE
Special Names Appear Blank

### ISSUE THREAD

**Issue by Deantwo** [StrangeLoopGames/EcoIssues#8261]
**Version:** 0.7.5.1 beta
**Server:** Centration Eco Server ([95.154.229.150:3110](http://95.154.229.150:3111/))
**Mods:** None

A friend on my server has a special name, `𝕲𝖑𝖔𝖗𝖎𝖔𝖚𝖘𝕭𝖎𝖗𝖇`, and the name appears as blank on the server we play on. He has told me that the name can appear on some other servers, so maybe it is a server operating system specific issue or something, but I have not seen this.

I am able to copy-paste the name off the website and into the game in other to create a link to his name, but tooltip always shows him as `Not Found`.
![image](https://user-images.githubusercontent.com/2676134/42133170-75051b42-7d25-11e8-837e-dd6bd0f80cad.png)
![image](https://user-images.githubusercontent.com/2676134/42133178-93632d7c-7d25-11e8-88c1-d603c5a795e7.png)

Trying to give him reputation resulted in an error message:
```
Server encountered an exception:

<size=60.00%>Exception: InvalidOperationException
Message:Reputation of null user requested.
Source:Eco.Gameplay

System.InvalidOperationException: Reputation of null user requested.
   at Eco.Gameplay.Economy.ReputationManager.OpenGiveRepUI(Player player, String targetUser)
   at Eco.Gameplay.Systems.TextLinks.TextLinkManager.OnLinkClicked(Player player, String linkId)</size>
```

**Comment by MarjieAVolk** [comment]
Related to https://github.com/StrangeLoopGames/EcoIssues/issues/8262

**Comment by MarjieAVolk** [comment]
So I just checked this out, and I found that when the server sent over the string '𝕲𝖑𝖔𝖗𝖎𝖔𝖚𝖘𝕭𝖎𝖗𝖇' in the form of a parameter in a link tag, the client then later gave back a string that was a bunch of Korean characters.  That seems quite odd.  But, that is just a novelty I thought I would share.

@Nanders has already fixed the issue with the name appearing blank - unsupported characters will now appear as '?'.

I am going to fix the link issue here - by using steam id/slg id as the identifier in the link, rather than the username.  Should be doing that anyway.

All this will be released probably not until 8.0, so not in the next release, but the one after
